src/awkward/highlevel.py

```diff
@@ -326,9 +338,20 @@ def __init__(
         if behavior is not None and not isinstance(behavior, Mapping):
             raise TypeError("behavior must be None or a mapping")
 
-        if attrs is not None and not isinstance(attrs, Mapping):
+        if attrs is not None and not isinstance(attrs, dict):
```

Comment:
`typing.Mapping` does not support `__setitem__` which I need in order to set named axis. I think the correct type annotation would then be `typing.MutableMapping` instead of `dict`?